scripts/build/tools/upload_to_s3.py

```diff
@@ -97,8 +98,11 @@ def get_files_to_upload(base_dir, regex, search_subdirectories):
     return regex_files_to_upload
 
 
-def s3_upload_file(client, file, bucket, key_prefix=None, extra_args=None, max_retry=1):
-    key = file if key_prefix is None else f'{key_prefix}/{file}'
+def s3_upload_file(client, base_dir, file, bucket, key_prefix=None, extra_args=None, max_retry=1):
+    # replicate the local folder structure relative to search root in the bucket path
+    s3_file_path = pathlib.Path(file).relative_to(base_dir).as_posix()
```

Comment:
You should put this in the a try except block as the relative_to command will toss an [exception](https://docs.python.org/3/library/pathlib.html?highlight=pathlib#pathlib.PurePath.relative_to) if the `file` is not within the `base_dir`

Reply:
Shouldn't be possible, but will add anyway just in case.